Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -465,23 +466,37 @@ DirectoryBlobReader::DirectoryBlobReader(
       auto type = m_wrapped_volume->GetPartitionType(partition);
       if (type)
       {
-        partitions.emplace_back(
-            DirectoryBlobPartition(m_wrapped_volume.get(), partition, m_is_wii, nullptr, nullptr),
-            static_cast<PartitionType>(*type));
+        partitions.emplace_back(DirectoryBlobPartition(m_wrapped_volume.get(), partition, m_is_wii,
+                                                       nullptr, nullptr, this),
+                                static_cast<PartitionType>(*type));
       }
     }
 
     SetPartitions(std::move(partitions));
   }
 }
 
+DirectoryBlobReader::DirectoryBlobReader(const DirectoryBlobReader& rhs)
+    : m_gamecube_pseudopartition(rhs.m_gamecube_pseudopartition),
+      m_nonpartition_contents(rhs.m_nonpartition_contents), m_encryption_cache(this),
+      m_is_wii(rhs.m_is_wii), m_encrypted(rhs.m_encrypted), m_data_size(rhs.m_data_size),
+      m_wrapped_volume(rhs.m_wrapped_volume ?
+                           CreateDisc(rhs.m_wrapped_volume->GetBlobReader().CopyReader()) :
+                           nullptr)
+{
+  for (auto entry : rhs.m_partitions)
+  {
+    m_partitions.emplace(entry.first, entry.second);
+  }
```

Comment:
Is there a reason why you have to emplace each partition manually? It seems to me like putting `m_partitions(rhs.m_partitions)` in the constructor initializer list should work.

Reply:
For _some reason_ I seem to have gotten it in my head that STL collections do _not_ deep copy and it's making me wonder if my education some twenty years ago misled me on it or if this genuinely has changed since my teachers harped me on it.

Reply:
The STL collections call the copy constructor for the objects they contain. So whether you get a deep copy depends on how the copy constructor is written.

Reply:
Oh, and if you put pointers inside a container instead of putting the objects directly inside the container, then you will indeed not get a deep copy. That's one of the many reasons why using raw pointers can be a hazard.